substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jni/functions/JNIFunctions.java

```diff
@@ -363,7 +367,15 @@ static int RegisterNatives(JNIEnvironment env, JNIObjectHandle hclazz, JNINative
         for (int i = 0; i < nmethods; i++) {
             JNINativeMethod entry = (JNINativeMethod) p;
             CharSequence name = Utf8.wrapUtf8CString(entry.name());
+            if (name == null) {
+                throw new NoSuchMethodError("<INVALID-NAME>");
```

Comment:
Please change to: `"Method name at index " + i + " is not a valid UTF-8 string or NULL"`